### PR TITLE
chore(remap, external docs): Add note about data model in the remap transform

### LINKED ISSUE #5667: Document Remap's handling of logs vs. metrics
We need to update the CUE schemas for Remap to indicate that it supports both logs and metrics *and* how it handles both. In addition, issue #5626 has pointed that there are important restrictions on metric fields that TRL can recognize, and there may be analogous subtleties for handling logs.

### PR DESCRIPTION
**Pull request by binarylogic**
Closes #5667

### CODE REVIEW

File: docs/reference/components/transforms/lua.cue [hunk 1]
```diff
@@ -38,7 +38,7 @@ components: transforms: lua: {
 		requirements: []
 		warnings: [
 			"""
-			The `lua` transform is ~60% slower than the [`remap` transform](\(urls.vector_remap_transform)), so we
+			The `lua` transform is ~60% slower than the [`remap` transform](\(urls.vector_remap_transform)); we
 			recommended that you use the `remap` transform whenever possible. The `lua` transform is
```
Comment (lucperkins) [inline docs/reference/components/transforms/lua.cue:42]: Not related to this PR, but s/recommended/recommend

File: docs/reference/remap.cue [hunk 2]
```diff
@@ -225,4 +225,15 @@ remap: #Remap & {
 				"""
 		},
 	]
+
+	how_it_works: {
+		event_data_model: {
+			title: "Event Data Model"
+			body:  """
+				The `remap` transform works with both log and metric events. The data model for log events matches
```
Comment (lucperkins) [inline docs/reference/remap.cue:233]: I'd suggest:

You can use the `remap` transform with both log and metric events. Log events in the `remap` transform correspond directly to Vector's log schema, which means that the transform has access to the whole event. With metric events, however, the `remap` transform only has access to the event's tags.

File: docs/reference/components/transforms/lua.cue (hunk 1 above)
Comment (JeanMertz) [inline docs/reference/components/transforms/lua.cue:42]:
```suggestion
			recommend that you use the `remap` transform whenever possible. The `lua` transform is
```

**Review by JeanMertz — APPROVED** [review]

**Comment by jszwedko** [comment]
@binarylogic this was failing CI checks which are now failing on master. I'll address, but it'd be good to check failures before merging PRs.

**Comment by jszwedko** [comment]
nvm, I see you already fixed it in https://github.com/timberio/vector/commit/8ed9267990526431aaa3a65806432770859bbebb 😄

**Pull request merged**